docs/examples/junit4/generic/src/test/java/org/testcontainers/containers/startupcheck/StartupCheckStrategyTest.java

```diff
@@ -84,9 +84,9 @@ public void testCommandIsExecuted() {
         @SneakyThrows
         @Test
         public void testCommandIsExecuted() {
-            waitForHello(bboxWithMinimumDuration);
-
             assertThat(bboxWithMinimumDuration.isRunning()).isTrue();
+
+            waitForHello(bboxWithMinimumDuration);
```

Comment:
I think it was wrong previously: once `waitForHello` returns, `sleep && echo` would complete and the container should stop. So it is wrong to expect `isRunning()` to return `true` here.

It worked fine before on Linux probably because the check was faster than containers' cleanup but I got a flaky behaviour on Windows

Reply:
I think you are right. Do you remember your original idea here @rnorth?

Reply:
I wasn't the original author, but it looks to me as though the original version was indeed prone to a race condition. It's quite surprising that it's taken so long for this to be detected.